tests/test_ensemble_tools.py

```diff
@@ -0,0 +1,163 @@
+"""Tests for the ensemble_tools utilities."""
+
+import itertools
+import re
+import shutil
+from pathlib import Path
+
+import pytest
+from defusedxml import ElementTree
+from py.path import local
+
+# from plugins.FabNESO.utils.make_sweep_dir
+from FabNESO.ensemble_tools import create_dict_sweep, create_dir_tree, edit_parameters
+
+
+def test_edit_parameters(tmpdir: local) -> None:
+    """Test the edit_parameters method of ensemble_tools."""
+    parameter_to_test = "particle_initial_velocity"
+    parameter_test_value = 10.0
+    temp_conditions_path = Path(tmpdir) / "conditions.xml"
+    shutil.copyfile(
+        Path("config_files") / "two_stream" / "conditions.xml", temp_conditions_path
+    )
+
+    edit_parameters(temp_conditions_path, {parameter_to_test: parameter_test_value})
+
+    assert _check_parameter_in_conditions(
+        temp_conditions_path, parameter_to_test, parameter_test_value
+    )
+
+
+def _check_parameter_in_conditions(
+    conditions_file_name: Path, parameter_name: str, expected_value: float
+) -> bool:
+    """Return True if parameter_name has approx expected_value."""
+    data = ElementTree.parse(conditions_file_name)
+    root = data.getroot()
+    conditions = root.find("CONDITIONS")
+    if conditions is None:
+        msg = f"Failed to find CONDITIONS in the file {conditions_file_name}"
+        raise ValueError(msg)
+    parameters = conditions.find("PARAMETERS")
+    if parameters is None:
+        msg = f"Failed to find PARAMETERS in the \
+        CONDITIONS node of {conditions_file_name}"
```

Comment:
```suggestion
        msg = (
            f"Failed to find PARAMETERS in the CONDITIONS node of "
            f"{conditions_file_name}"
        )
```
[PEP 8 guidance for wrapping long lines is](https://peps.python.org/pep-0008/#maximum-line-length)

> The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.